cypress/e2e/editing/editing.spec.ts

```diff
@@ -55,14 +60,269 @@ describe("Editing", () => {
+
+        // Take a snapshot
+        cy.get(".mx_Dialog .mx_MessageEditHistoryDialog").percySnapshotElement("Message edit history dialog", {
+            percyCSS,
+            widths: [704], // See: .mx_Dialog_fixedWidth max-width
+        });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Click "Remove" button on MessageActionBar
+            cy.get(".mx_MessageEditHistoryDialog").within(() => {
+                // Assert that the edited message is rendered
+                cy.get("li:nth-child(2) .mx_EventTile").within(() => {
+                    // Click the "Remove" button
+                    cy.get(".mx_EventTile_line")
+                        .realHover()
+                        .contains(".mx_AccessibleButton", "Remove")
+                        .click({ force: false });
+                });
+            });
+
+            // Do nothing and close the dialog to confirm that the message edit history dialog is rendered
+            cy.get(".mx_TextInputDialog").within(() => {
+                cy.get("[aria-label='Close dialog']").click();
+            });
+
+            // Assert that the message edit history dialog is rendered again after it was closed
+            cy.get(".mx_MessageEditHistoryDialog").within(() => {
+                // Assert that the edited message is rendered again
+                cy.get("li:nth-child(2) .mx_EventTile").within(() => {
+                    cy.get(".mx_EventTile_content").within(() => {
+                        cy.get(".mx_EventTile_body").should("have.text", "Meassage");
+                    });
+
+                    // Click the "Remove" button again
+                    cy.get(".mx_EventTile_line")
+                        .realHover()
+                        .contains(".mx_AccessibleButton", "Remove")
+                        .click({ force: false });
+                });
+            });
+
+            // This time remove the message really
+            cy.get(".mx_TextInputDialog").within(() => {
+                cy.get(".mx_TextInputDialog_input").type("This is a test."); // Reason
+                cy.contains("[data-testid='dialog-primary-button']", "Remove").click();
+            });
+
+            // Assert that the message edit history dialog is rendered again
+            cy.get(".mx_MessageEditHistoryDialog").within(() => {
+                // Assert that the date is rendered
+                cy.get("li:nth-child(1) .mx_DateSeparator").within(() => {
+                    cy.get("h2").should("have.text", "Today");
+                });
+
+                // Assert that the original message is rendered on the dialog
+                cy.get("li:nth-child(2) .mx_EventTile").within(() => {
+                    cy.get(".mx_EventTile_content .mx_EventTile_body").should("have.text", "Message");
+                });
+
+                // Assert that the edited message is gone
+                cy.get("li:nth-child(3) .mx_EventTile").should("not.exist");
```

Comment:
It would probably be better to check for the non-existence of „Meassage“ here.
The test already heavily relies on CSS structure.
At least in this particular place it can be avoided.

Reply:
I'd agree, let me check how it should be changed...